### ISSUE TITLE
README MR simulation equation

### ISSUE THREAD

**Issue by evaalonsoortiz** [neuropoly/Susceptibility-Separation-Phantom#2]
The README has the following equation for MR GRE signal modeling:

$$S=M_0\sin(\alpha)\frac{1-e^{-TR.R_1}}{1-\cos(\alpha)e^{-TR.R_1}}e^{-TE.(R_2+D_r(|\chi^+|+|\chi^-|))+i(\Phi_0+2\pi{\gamma}TE.B_0(D(\chi^++\chi^-)))}$$

I'm concerned about the last term:
$$e^{i(\Phi_0+2\pi{\gamma}TE.B_0(D(\chi^++\chi^-)))}$$

Ignoring the initial phase offset, and the pos/neg susceptibility, you have:

$$e^{2i\pi{\gamma}TE.B_0.D.\chi}$$

The periodic component of the signal should be represented by:
$$e^{i\Delta\omega TE}$$

where
$$\Delta \omega = \gamma \Delta B$$
and
$$\Delta B = D \star \Delta \chi $$

resulting in:
$$e^{i{\gamma}TE.D \star \Delta \chi}$$

In your equation, why is there a B0 and a 2pi? Why is there a multiplication and not a convolution? Am I missing something?

In your code, I think things match the equations that I derived here. 

https://github.com/neuropoly/Susceptibility-Separation-Phantom/blob/6733f8c790cfba44fb7e3e2f2e2aa6e199ba371a/func/GRESimulation.m#L76-L77

There's no gamma in the above code though, where can I find the definition? What units is `field` in?

**Comment by Danirid** [comment]
In my code, I also multiply by $B_0$, $2\pi$, and $\gamma$, it's not shown in the GRE signal equation but rather here:
https://github.com/neuropoly/Susceptibility-Separation-Phantom/blob/6733f8c790cfba44fb7e3e2f2e2aa6e199ba371a/func/DataSimulationFunction.m#L92

Regarding the  $2\pi$ it is usually added when computing the acquired phase as seen here in this equation:
![image](https://github.com/neuropoly/Susceptibility-Separation-Phantom/assets/77506715/891e13df-cd3d-478f-893f-2e541cd3a723)

Also in the $\chi -separation$ article when writing the signal equation they included a $2\pi$ factor:
![image](https://github.com/neuropoly/Susceptibility-Separation-Phantom/assets/77506715/87e68fc3-1ebe-4ee9-badf-8b9bd3bc14fb)

I think it comes from the fact that $w$= $2\pi$ x $f$

And regarding $B_0$ I've some resources that include $B_0$ when writing the field distribution equation:
![image](https://github.com/neuropoly/Susceptibility-Separation-Phantom/assets/77506715/a9920b19-a71a-46aa-b1d9-f83d9fddbb35)
But I've also seen articles that do not include $B_0$

**Comment by evaalonsoortiz** [comment]
Ok, I missed the B0. You're correct:
$$\Delta B = B_0 \cdot (d(r) \star \chi)$$
If you see equations without the B0 it is because the deltaB0 is expressed in ppm.

For the 2pi, your response doesn't really address my question. After looking in your code, I found that your definition of gamma is:
$$\gamma = 42.58$$
there should be units in a comment beside the definition. This value is in units of MHz/T. 
1. Are you accounting for the missing 10^6?
2. This is why you have a 2pi in your equations:
$$\gamma = 2.86\cdot 10^8 \left[ \frac{rad}{s \cdot T} \right] $$
whereas 
$$\rlap{\gamma}- = \frac{\gamma}{2\pi} = 42.6\cdot 10^6 \left[ \frac{1}{s \cdot T} \right] $$

**Comment by Danirid** [comment]
Oh yes, that's correct I did add the $2\pi$. In the original QSM phantom $\gamma$ was not multiplied by $2\pi$.
Regarding the units, the missing $\ 10^6$ has to be accounted for in the code. Because I ran the code 2 times ( with and without the $\ 10^6$):

 
![image](https://github.com/neuropoly/Susceptibility-Separation-Phantom/assets/77506715/ed1b0f04-29b8-47f1-ab50-4507fe1b9775)

When I added the $\ 10^6$ (image on the left) I got random phase values. But when excluding the $\ 10^6$ (image on the right) the phase looked great. I will have a closer look at the code maybe we are missing something.

**Comment by evaalonsoortiz** [comment]
Indeed, it must be accounted for somewhere ... let me know here once you've identified it.
